src/App.js

```diff
@@ -1,19 +1,49 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const [likeCount, setLikeCount] = useState(0);
+
+  const handleLikeUpdate = (id) => {
+    setMessages(messages => {
+      return messages.map(message => {
+        if (id === message.id) {
+          return {
+            ...message,
+            liked: !message.liked,
+          }
+        } else {
+          return message;
+        }
+      });
+    });
+  };
+
+  const updateLikeCount = ((liked) => {
+    if (liked) {
+      setLikeCount(() => likeCount + 1);
+    } else{
+      setLikeCount(() => likeCount - 1);
+    }
+  });
```

Comment:
👀 If we go with one of the approaches to calculate the liked count from the message data, then we don't need this function to manage the liked count state.